src/main/java/org/apache/sysml/api/mlcontext/Matrix.java

```diff
@@ -160,11 +247,45 @@ public BinaryBlockMatrix toBinaryBlockMatrix() {
 	 * @return the matrix metadata
 	 */
 	public MatrixMetadata getMatrixMetadata() {
-		return new MatrixMetadata(matrixObject.getMatrixCharacteristics());
+		return matrixMetadata;
 	}
 
 	@Override
 	public String toString() {
-		return matrixObject.toString();
+		if (matrixObject != null) {
+			return matrixObject.toString();
+		} else if (matrixMetadata != null) {
+			return matrixMetadata.toString();
+		} else {
+			return super.toString();
+		}
```

Comment:
nice to have comment: could we make this easier to read via something like "return (matrixObject !=null) ? matrixObject.toString() : (matrixMetadata !=null ) ? matrixMetadata.toString() : super.toString();"?